src/llmcompressor/modifiers/awq/base.py

```diff
@@ -0,0 +1,746 @@
+    in one-shot and not during training. Activation ranges are determined by running a
+    small set of calibration data through the model.
+
+    example recipe:
+    ```yaml
+    AWQModifier:
+      bits: 4
+      mappings:
+        - smooth_layer: "re:.*self_attn_layer_norm"
+          balance_layers: ["re:.*q_proj", "re:.*k_proj", "re:.*v_proj"]
+        - smooth_layer: "re:.*final_layer_norm"
+          balance_layers: ["re:.*fc1"]
+      ]
+      ignore: ["model.decoder.final_layer_norm"]
+    ```
+
+    :param mappings: list activation layers to smooth, and which layers to
+        scale the output such that activations are smoothed.
+        Each entry of the mapping list should be a list itself, in which the first
+        entry is a list of layers who share the same input activation (the one to be
+        to smoothed) and the second entry is the layer whose output is scaled to
+        achieve the smoothing.
+        If regex is used, it matches layers with the largest overlap in module name.
+    :param ignore: list of layers to ignore, even if they match a regex in mappings.
+        It should match the name of layers whose outputs are scaled to achieve
+        smoothing (the second entry of the mappings list).
+    :param num_calibration_steps: number of samples to use for calibration, or None to
+        use the whole dataset
+    :param calibration_function: optional function to use for the forward pass, or None
+        to use the default tensor_module_forward
+    :param group_size: number of weights to group together for scaling
+    :param max_chunk_memory: maximum memory to use for each chunk of input activations
+    :param bits: number of bits to quantize the weights to
+    :param symmetric: whether to use symmetric quantization
+    :param duo_scaling: whether to use duo scaling, which uses both input activations
+        and weights to determine the scaling factor
+    :param apply_clip: whether to apply clipping to the weights after scaling
+    """
+
+    # Allow arbitrary types because AWQMapping has fields of type torch.nn.Module
+    model_config: ConfigDict = ConfigDict(arbitrary_types_allowed=True)
+
+    mappings: List[AWQMapping] = DEFAULT_AWQ_MAPPINGS
+    ignore: Optional[List[str]] = None
+    num_calibration_steps: Optional[int] = None
+    calibration_function: Optional[Callable] = None
+    group_size: int = 128
+    max_chunk_memory: int = 1024 * 1024 * 1024
+    bits: int = 4
+    symmetric: bool = True
+    duo_scaling: bool = True
+    apply_clip: bool = True
+
+    resolved_mappings_: Optional[List[ResolvedMapping]] = None
+    scales_: Optional[Dict] = None
+    module_kwargs_: Optional[Dict] = None
+
+    def on_initialize(self, state: State, **kwargs) -> bool:
+        """
+        Initialize and run AWQ on the given state
+
+        :param state: state to run AWQ on
+        :return: True on a successful run, False otherwise
+        """
+        if not (self.end is None or self.end == -1):
```

Comment:
I would consider removing these checks, as they're not necessary

Reply:
removed